cunumeric/utils.py

```diff
@@ -45,6 +45,25 @@
     np.complex128: ty.complex128,
 }
 
+CUNUMERIC_TYPE_MAP = {
+    "bool": ty.bool_,
+    "int8": ty.int8,
+    "int16": ty.int16,
+    "int32": ty.int32,
+    "int": ty.int64,  # np.int is int
+    "int64": ty.int64,
+    "uint8": ty.uint8,
+    "uint16": ty.uint16,
+    "uint32": ty.uint32,
+    "uint64": ty.uint64,  # np.uint is np.uint64
+    "float16": ty.float16,
+    "float32": ty.float32,
+    "float": ty.float64,
+    "float64": ty.float64,
+    "complex64": ty.complex64,
+    "complex128": ty.complex128,
+}
```

Comment:
Yes, probably we didn't need it before. Or, maybe there is a better way of getting `ty` type from python type string
@manopapad ?

Reply:
It seems that this table is used later to map python types to Legate.Core (i.e. pyarrow) types. In that case the only entries here that matter are "bool", "int" and "float", since these are the only names of built-in python arithmetic types (there is also "complex", but that's not present here). I assume then we could change this dict to map `type` to `pa.lib.DataType`, possibly add "complex" (mapped to `ty.complex128`) and use it directly, instead of going through `__name__`:

```
         for a in self._scalar_args:
             dtype = convert_to_cunumeric_dtype(type(a))
```

Reply:
Thank you, @manopapad . I have just pushed the suggested change.